Apps/Src/ReadCarCAN.c

```diff
@@ -31,7 +31,7 @@ void Task_ReadCarCAN(void *p_arg) {
         // Normal task countinues here
 
         // Check if BPS sent us a message
-        if (CANbus_Read(buffer) == SUCCESS) {
+        if (CANbus_Read(buffer,CAN_NON_BLOCKING) == SUCCESS) {
             long msg = *((long *)(&buffer[0]));
             if (msg == 0) {
```

Comment:
How are we sure that this message come from the BPS? How are we sure which message from the BPS it is?

Reply:
I also have problems with this function in general, and think that the logic of it may need to be thought through again.